File: src/wesoga/entities/EntityModel.java
package wesoga.entities;

import wesoga.textures.Textures;

public final class EntityModel {
	public final int northTexture;
	public final int southTexture;
	public final int eastTexture;
	public final int westTexture;
	public final int topTexture;
	public final int botTexture;

	public final double size;
	public final double height;

	public EntityModel(byte[] north, byte[] south, byte[] east, byte[] west, byte[] top, byte[] bot, double size,
			double height) {
		northTexture = Textures.textureID(north);
		southTexture = Textures.textureID(south);
		eastTexture = Textures.textureID(east);
		westTexture = Textures.textureID(west);
		topTexture = Textures.textureID(top);
		botTexture = Textures.textureID(bot);

		this.size = size;
		this.height = height;
	}

	@Override
	public final boolean equals(Object obj) {
		return this == obj;
	}
}

File: src/resources/web/entity_model.ts
class EntityModel {
	constructor(northTexture: number, southTexture: number, eastTexture: number, westTexture: number, topTexture: number, botTexture: number, size: number, height: number) {
		this.northTexture = northTexture;
		this.southTexture = southTexture;
		this.eastTexture = eastTexture;
		this.westTexture = westTexture;
		this.topTexture = topTexture;
		this.botTexture = botTexture;

		this.size = size;
		this.height = height;
	}

	public northTexture: number;
	public southTexture: number;
	public eastTexture: number;
	public westTexture: number;
	public topTexture: number;
	public botTexture: number;

	public size: number;
	public height: number;
}
